core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -5,8 +5,8 @@ import akka.http.scaladsl.model.headers.OAuth2BearerToken
 import akka.stream.Materializer
```

Comment:
Sorry for the noise in this file. Recommend viewing side-by side.
Main changes:
- Pull out common methods into `WorkspaceSupport`
- Refactor copy + clone to remove work duplication
- Flatten deeply-nested methods that were affected by ^